site/gatsby-site/src/components/discover/Actions.js

```diff
@@ -9,27 +9,59 @@ import {
   faFlag,
   faHashtag,
 } from '@fortawesome/free-solid-svg-icons';
-import { FIND_REPORT, UPDATE_REPORT } from '../../graphql/reports';
+import { FIND_REPORT, UPDATE_REPORT, LOG_REPORT_HISTORY } from '../../graphql/reports';
 import { useMutation, useQuery } from '@apollo/client';
 import { Trans, useTranslation } from 'react-i18next';
 import CustomButton from '../../elements/Button';
 import { Modal } from 'flowbite-react';
+import { useUserContext } from 'contexts/userContext';
+import { format, getUnixTime } from 'date-fns';
+import { transformReportData } from '../../utils/reports';
 
 function FlagModalContent({ reportNumber }) {
+  const { user } = useUserContext();
+
   const { data } = useQuery(FIND_REPORT, {
     variables: { query: { report_number: reportNumber } },
   });
 
-  const [flagReport, { loading }] = useMutation(UPDATE_REPORT, {
-    variables: {
-      query: {
-        report_number: reportNumber,
-      },
-      set: {
-        flag: true,
+  const [flagReportMutation, { loading }] = useMutation(UPDATE_REPORT);
+
+  const [logReportHistory] = useMutation(LOG_REPORT_HISTORY);
+
+  const flagReport = async () => {
+    const now = new Date();
+
+    const updated = {
+      flag: true,
+      date_modified: format(now, 'yyyy-MM-dd'),
+      epoch_date_modified: getUnixTime(now),
+    };
+
+    await flagReportMutation({
+      variables: {
+        query: {
+          report_number: reportNumber,
+        },
+        set: {
+          ...updated,
+        },
       },
-    },
-  });
+    });
+
+    const report = transformReportData(data.report, user);
+
+    const logReport = {
+      ...report,
+      ...updated,
+    };
+
+    // Set the user as the last modifier
+    logReport.modifiedBy = user && user.providerType != 'anon-user' ? user.id : '';
```

Comment:
This because `anon`s can flag and it counts as a change?

Reply:
Yes, you can flag without login. In that case we log the change with the empty `modifiedBy` field